pkg/ingester/ingester_test.go

```diff
@@ -588,7 +588,7 @@ func Test_InMemoryLabels(t *testing.T) {
 
 	res, err = i.Label(ctx, &logproto.LabelRequest{})
 	require.NoError(t, err)
-	require.Equal(t, []string{"bar", "foo"}, res.Values)
+	require.Equal(t, []string{"__name__", "bar", "foo"}, res.Values)
```

Comment:
Do I read that right, that there were cases that label queries on the ingester did _not_ contain the `__name__` label, even though labels were present?
If so, I think it would be good to also add this in the PR description.

Reply:
There were. It was relying on the fact that a `__name__` label always came from the store: whether there were series or not. This was actually found as a result of the new integration tests.

I'll add it to the PR description